auth.go

```diff
@@ -15,13 +15,16 @@ import (
 	"crypto/sha256"
 	"crypto/x509"
 	"encoding/pem"
+	"fmt"
 	"sync"
 )
 
 // server pub keys registry
 var (
-	serverPubKeyLock     sync.RWMutex
-	serverPubKeyRegistry map[string]*rsa.PublicKey
+	serverPubKeyLock           sync.RWMutex
+	serverPubKeyRegistry       map[string]*rsa.PublicKey
+	credentialProviderLock     sync.RWMutex
+	credentialProviderRetistry map[string]CredentialProviderFunc
```

Comment:
`credentialProviderRetistry` looks like it's typo'd throughout. (Should be `credentialProviderRegistry` instead)